torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -208,12 +208,64 @@ class TensorExprFuser {
   }
 
  private:
+  Node* aliasingSafeSubgraphMerge(
+      Node* to_merge,
+      std::function<Node*(std::unordered_map<Value*, Value*>&)> merge_fn) {
+    // When we merge a node into a subgraph, the new subgraph outputs
+    // have the same aliasing properties as the original node's outputs.
+    // Here we create a placeholder node, transfer the aliasing properties
+    // to the placeholder, execute the merge, and transfer the aliasing
+    // properties to the appropriate fusion group outputs
+    Node* placeholder_node =
+        graph_->insertNode(graph_->create(prim::Uninitialized, 0));
+    std::vector<Value*> existing_values;
+    for (size_t i = 0; i < to_merge->outputs().size(); ++i) {
+      Value* existing = to_merge->outputs().at(i);
+      Value* new_value = placeholder_node->insertOutput(i)->copyMetadata(
+          to_merge->outputs().at(i));
+      aliasDb_->replaceWithNewValue(existing, new_value);
+      existing_values.push_back(existing);
+    }
+    std::unordered_map<Value*, Value*> vmap;
+    Node* fusion_group = merge_fn(vmap);
```

Comment:
There is a single user of this function and for that user `merge_fn` = `SubgraphUtils::mergeNodeIntoSubgraph`. Do we really need to use a lambda here? Feels like we could be over-engineering it here.

Reply:
It's also `getOrCreateTensorExprSubgraph`